### PR TITLE
Correct sxhash to spec, test for this problem and other hash-table-issues

### PR DESCRIPTION
**Pull request by kpoeck**
Solves #558 (retested with ansi-tests)

### CODE REVIEW

File: src/core/primitives.cc [hunk 1]
```diff
@@ -1565,12 +1565,17 @@ CL_DEFUN T_sp cl__type_of(T_sp x) {
...
+CL_DEFUN Fixnum_sp cl__sxhash(T_sp obj) {
   if (obj.nilp())
     return make_fixnum(1);
   HashGenerator hg;
   clasp_sxhash(obj, hg);
-  return Integer_O::create(hg.hash());
+  gc::Fixnum hash = hg.hash();
+  if (hash < 0)
+    hash = - hash;
+  if (hash > MOST_NEGATIVE_FIXNUM)
```
Comment (drmeister) [inline src/core/primitives.cc]: Won't this always be true given the if (hash <0) hash = -hash?

Comment (kpoeck) [reply src/core/primitives.cc]: In current clasp:
(sxhash #*11) -> -355494570821960769
so I don't think that if (hash <0) hash = -hash holds in the current implementation

Comment (Bike) [reply src/core/primitives.cc]: It might be better to figure out where the negative value is coming from.

**Comment by drmeister** [comment]
Great!

**Pull request merged**